barretenberg/cpp/src/barretenberg/vm/avm/trace/bytecode_trace.hpp

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include <utility>
+
+#include "barretenberg/vm/avm/trace/common.hpp"
+#include "barretenberg/vm/avm/trace/execution_hints.hpp"
+namespace bb::avm_trace {
+
+class AvmBytecodeTraceBuilder {
+  public:
+    struct BytecodeTraceEntry {
+        // Calculate the bytecode hash
+        FF packed_bytecode{};
+        FF running_hash{};
+        // This is the length in fields, not bytes - max 1000 fields
+        uint16_t bytecode_length_remaining = 0;
+
+        // Derive the class Id
+        FF class_id{};
+
+        // Derive the contract address
+        FF contract_address{};
+    };
+    // These interfaces will change when we start feeding in more inputs and hints
+    AvmBytecodeTraceBuilder(const std::vector<uint8_t>& contract_bytecode, const ExecutionHints& hints)
+    {
+        // Do we need to pad contract_bytecode to be 31bytes?
+        std::vector<std::vector<uint8_t>> all_contracts_bytecode{ contract_bytecode };
```

Comment:
You might want to do this at deserialization time, and just pass whatever this class needs. In particular, this would also avoid doing work in this constructor, which in general is something to be avoided.